app/src/main/java/com/android/sample/ui/swipePage/SwipePage.kt

```diff
@@ -160,6 +161,163 @@ fun RecipeDisplay(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier =
+                Modifier.testTag("filterRow")
+                    .fillMaxWidth()
+                    .weight(1f)
+                    .horizontalScroll(rememberScrollState())) {
+              // Star Icon (fixed size, no weight needed)
+              var displayTimeRange by remember { mutableStateOf(!filter.timeRange.isLimited()) }
+              var displayPriceRange by remember { mutableStateOf(!filter.priceRange.isLimited()) }
+              var displayDifficulty by remember {
+                mutableStateOf(filter.difficulty != Difficulty.Undefined)
+              }
+              var displayCategory by remember { mutableStateOf(filter.category != null) }
+
+              if (displayTimeRange) {
+                // Display the time range chip
+                InputChip(
+                    modifier = Modifier.testTag("timeRangeChip"),
+                    selected = false,
+                    onClick = {},
+                    label = {
+                      Text(
+                          text =
+                              "${filter.timeRange.min.toInt()} - ${filter.timeRange.max.toInt()} min",
+                          style = MaterialTheme.typography.bodySmall,
+                          color = MaterialTheme.colorScheme.onSecondary,
+                      )
+                    },
+                    trailingIcon = {
+                      Icon(
+                          Icons.Filled.Close,
+                          contentDescription = TIME_RANGE_INPUT_DESCRIPTION,
+                          modifier =
+                              Modifier.testTag("timeFilterDelete")
+                                  .size(InputChipDefaults.IconSize)
+                                  .clickable {
+                                    displayTimeRange = false
+                                    recipesViewModel.updateTimeRange(
+                                        recipesViewModel.filter.value.timeRange.minBorn,
+                                        recipesViewModel.filter.value.timeRange.maxBorn)
+                                  },
+                          tint = MaterialTheme.colorScheme.onSecondary)
+                    })
+
+                Spacer(modifier = Modifier.width(8.dp))
+              }
+              if (displayPriceRange) {
+                // Display the time range chip
+                InputChip(
+                    modifier = Modifier.testTag("priceRangeChip"),
+                    selected = false,
+                    onClick = {},
+                    label = {
+                      Text(
+                          text =
+                              "${filter.priceRange.min.toInt()} - ${filter.priceRange.max.toInt()} $",
+                          style = MaterialTheme.typography.bodySmall,
+                          color = MaterialTheme.colorScheme.onSecondary,
+                      )
```

Comment:
Couldn't this be extracted in a function, this would make the Page more concise and more readable.

Reply:
Done! And great idea, I’ve extracted it into a function to make the page cleaner and more readable.